_pytest/config.py

```diff
@@ -1108,29 +1110,39 @@ def get_common_ancestor(args):
                     common_ancestor = shared
     if common_ancestor is None:
         common_ancestor = py.path.local()
-    elif not common_ancestor.isdir():
+    elif common_ancestor.isfile():
         common_ancestor = common_ancestor.dirpath()
     return common_ancestor
 
+def get_dirs_from_args(args):
+    return [d for d in (py.path.local(x) for x in args if not
+                        str(x).startswith("-")) if d.exists()]
```

Comment:
The formatting of this makes it pretty hard to read. Could `d for d in` be aligned with `if d.exists()` more clearly?

Would splitting into two statements, or something like this be better:?
```python
return [d for d in (py.path.local(x) for x in args if not
                    str(x).startswith("-"))
        if d.exists()]
```